### ISSUE TITLE
Question about channel life cycle

### ISSUE THREAD

**Issue by hankai17** [yedf2/handy#103]
https://github.com/yedf2/handy/blob/c16cde9ae1c801ededf2555519bbab2af5496625/handy/poller.cc#L97
https://github.com/yedf2/handy/blob/c16cde9ae1c801ededf2555519bbab2af5496625/handy/conn.cc#L109
It seem's in channel's Callback delete channel.

**Comment by yedf2** [comment]
Yes, if the data stream in the tcp connection has finished, then the channel should be closed and deleted
